utils/atomic-data-publisher/src/errors.rs

```diff
@@ -0,0 +1,58 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum AtomicDataError {
+  #[error("Database error: {0}")]
+  DatabaseError(String),
```

Comment:
`thiserror` provides `#[from]` attribute that auto implements of much of the stuff you have below.

```rs
  #[error("Database error: {0}")]
  DatabaseError(#[from] sqlx::Error)
```

This also means we keep all the context of the error stack, rather than what we get from `.to_string()`.